docs/howtos/upgrading-nss-guide.md

```diff
@@ -5,12 +5,63 @@ Every month or so, a new version of NSS is [published](https://developer.mozilla
 
 Because it makes unit testing easier on Android, and helps startup performance on iOS, we compile NSS ourselves and link to it statically. Note that NSS is mainly used by Mozilla as a dynamic library and the NSS project is missing related CI jobs (iOS builds, windows cross-compile builds etc.) so you should expect breakage when updating the library (hence this guide).
 
-The build code is located in the [`libs/`](https://github.com/mozilla/application-services/tree/main/libs) folder.  
-The version string is located in the beginning of [`build-all.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-all.sh#L8-L11). For most NSS upgrades, the only action needed is to bump the version number in this file and update the downloaded archive checksum.  The actual build invocations are located in platform-specific script files (e.g. [`build-nss-ios.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-nss-ios.sh))
+## Updating the Version
 
-On top of that, we have built a safe Rust wrapper named [rc_crypto](https://github.com/mozilla/application-services/tree/main/components/support/rc_crypto) that links to NSS and makes these cryptographic primitives available to our components.
+The build code is located in the [`libs/`](https://github.com/mozilla/application-services/tree/main/libs) folder.
+
+The version string is located in the beginning of [`build-all.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-all.sh#L8-L11). For most NSS upgrades, the only action needed is to bump the version number in this file and update the downloaded archive checksum.  The actual build invocations are located in platform-specific script files (e.g. [`build-nss-ios.sh`](https://github.com/mozilla/application-services/blob/b0b3daa6580d04906fc53e9e479e8bebb464cf78/libs/build-nss-ios.sh)) but usually don't require any changes.
+
+Due to the complexity of the NSS build process, we currently use pre-built artifacts
+for running tests on MacOS and Win32 desktop machines (ref [#962](https://github.com/mozilla/application-services/issues/962)). You can regenerate these artifacts by taking advantage of the [NSS TryServer](https://wiki.mozilla.org/NSS:TryServer) infrastructure, with some patches to the NSS source:
+
+1. Get the source for NSS:
+    * `hg clone https://hg.mozilla.org/projects/nss`
+    * `cd nss`
+1. Apply [this patch](https://phabricator.services.mozilla.com/D30511) for Windows mingw build support, rebasing it on top
```

Comment:
oh god, help us all :(

Reply:
Yeah :-/

I actually have no idea how to test whether the resulting build is actually useable, but we don't currently ship Windows unittest builds anyway because of https://github.com/mozilla/application-services/issues/3917, so it probably doesn't matter.